packages/netlify/src/index.ts

```diff
@@ -112,13 +112,25 @@ async function writeNetlifyFrameworkConfig(config: AstroConfig, logger: AstroInt
 			.filter(Boolean as unknown as (pattern?: string) => pattern is string)
 	);
 
+	const headers = config.build.assetsPrefix
+		? undefined
+		: [
+				{
+					for: `${config.base}${config.base.endsWith('/') ? '' : '/'}${config.build.assets}/*`,
```

Comment:
What happens if base isn't set? Also maybe worth extracting to withTrailingSlash if such logic is used elsewhere